### PR TITLE
Adopt the Xanadu Sphinx Theme

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

Currently, all Xanadu OSS projects include their own version of the Xanadu Sphinx theme; however, the [Xanadu Sphinx Theme](https://github.com/XanaduAI/xanadu-sphinx-theme) repository is now publicly available and is the preferred way to access the Xanadu CSS theme and Sphinx directives.

**Description of the Change:**

* Deleted the docs/xanadu_theme directory.
* Updated docs/requirements.txt and docs/conf.py to reference and use the (centralized) Xanadu Sphinx Theme.
* Added a Python 3.10 classifier to setup.py.

**Benefits:**

* The size of this repository is smaller, meaning `git clone` will run faster.
* Updates to the Xanadu Sphinx theme no longer have to be manually ported to this repository.

**Possible Drawbacks:**
None.

**Related GitHub Issues:**
Depends on XanaduAI/xanadu-sphinx-theme#2.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -9,9 +9,9 @@ jobs:
   sphinx:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: josh146/sphinx-action@master
+    - uses: actions/checkout@v2
+    - uses: XanaduAI/sphinx-action@master
```
Comment (josh146) [inline .github/workflows/docs.yml:13]: 🙌

File: docs/conf.py [hunk 2]
```diff
@@ -251,39 +240,32 @@
...
+        }
+    ],
+
+    "border_colour": "#78909c",
+    "prev_next_button_colour": "#607d8b",
+    "prev_next_button_hover_colour": "#34515e",
+    "toc_marker_colour": "#78909c",
+    "table_header_background_colour": "#78909c",
+    "text_accent_colour": "#78909c",
+}
```
Comment (josh146) [inline docs/conf.py:271]: This is so great!

**Review by josh146 — APPROVED** [review]
Looks 💯 to me @Mandrenkov, although I didn't clone and attempt to build the theme by myself

File: docs/conf.py (hunk 2 above)
Comment (Mandrenkov) [reply docs/conf.py:271]: Awesome!

**Comment by Mandrenkov** [comment]
Thanks for the review, @josh146!

> Looks 💯 to me @Mandrenkov, although I didn't clone and attempt to build the theme by myself

Sounds good! I think it would be worthwhile to build the documentation locally for PL, PL-LQ, and SF though.

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -39,6 +39,10 @@
 
 ### Documentation
 
+* The centralized [Xanadu Sphinx Theme](https://github.com/XanaduAI/xanadu-sphinx-theme)
+  is now used to style the Sphinx documentation.
+  [(#23)](https://github.com/XanaduAI/xanadu-cloud-client/pull/23)
+
```
Comment (trbromley) [inline .github/CHANGELOG.md:45]: Minor point - @Mandrenkov are you listed as a contributor for this release?

**Review by trbromley — APPROVED** [review]
Nice!

File: .github/CHANGELOG.md (hunk 3 above)
Comment (Mandrenkov) [reply .github/CHANGELOG.md:45]: Good catch! 🤦‍♂️

**Pull request merged**